packages/sdk/src/marketplaces/approval.ts

```diff
@@ -0,0 +1,92 @@
+import { Signer } from "@ethersproject/abstract-signer";
+import { BaseProvider } from "@ethersproject/providers";
+import { NftSwapV4 } from "@traderxyz/nft-swap-sdk";
+
+import { Asset } from "../types";
+
+import type {
+  SwappableAssetV4,
+  UserFacingERC20AssetDataSerializedV4,
+  UserFacingERC721AssetDataSerializedV4,
+  UserFacingERC1155AssetDataSerializedV4,
+} from "@traderxyz/nft-swap-sdk";
+
+/**
+ * Approves asset for use on specified contract address.
+ * @param params Params for asset approval
+ */
+export async function approveAsset({
```

Comment:
Or just expose this method from `Trader` and call it, without redefining the methods here?

```typescript
  async approveAsset(asset: Asset | SwappableAssetV4): Promise<void> {
    return approveAsset(this.nftSwapSdk, this.address, asset);
  }
}

export async function approveAsset(
  nftSwapSdk: NftSwapV4,
  address: string,
  asset: Asset | SwappableAssetV4
): Promise<void> {
...
```